### ISSUE TITLE
Strange Error when building erroneous project again without changing

### ISSUE THREAD

**Issue by HerrNamenlos123** [chalet-org/chalet#341]
I still have the same simple project (using C++20 Modules) and it does not compile, because main.cc includes a header from a dependency it cannot find. Running `chalet build` results in:

```
󰁔 chalet build
>  Reading Build File [chalet.json] ... done (24ms)
>  Reading Microsoft (R) Visual C/C++ Environment Cache ... done
>  Configuring build ... done (1ms)
>  Resolving source file dependencies ... done (1ms)
>  C++ Compiler: Microsoft (R) Visual C/C++ version 19.37.32825 (VS 17.7.34221.43)
>  Target Architecture: x86_64-pc-windows-msvc
>  Strategy: Native
>  Configuration: Release

»  Build: cppgfx

   cppgfx: Target is up to date.

»  Build: inversekinematics.exe

   [1/1] build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.module.json
FAILED: C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/bin/HostX64/x64/cl.exe /nologo /c /source-charset:UTF-8 /execution-charset:UTF-8 /validate-charset /MP /TP /std:c++lat
est /experimental:module /stdIfcDir C:/Program Files/Microsoft Visual Studio/2022/Community/VC/Tools/MSVC/14.37.32822/ifc/x64 /ifcOutput build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.
ifc /sourceDependencies:directives build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.module.json /interface /Zi /FS /W3 /diagnostics:caret /sdl /O2 /Ob2 /Oi /EHsc /MD /GS /fp:precise /Gy 
/permissive- /Zc:wchar_t /Zc:forScope /Zc:inline /Fdbuild/x86_64-pc-windows-msvc_Release/obj.inversekinematics/vc143.pdb /external:W3 /Gd /Isrc /Ichalet_external/cppgfx-src/include /Ibuild/x86_64-pc-windows-m
svc_Release/ext.cppgfx/_deps/include /Ibuild/x86_64-pc-windows-msvc_Release/obj.inversekinematics /Ibuild/x86_64-pc-windows-msvc_Release/int /Fobuild/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/m
ain.cc.obj src/main.cc
C:\Users\zachs\Projects\inversekinematics\chalet_external\cppgfx-src\include\cppgfx/cppgfx.hpp(5,10): fatal error C1083: Cannot open include file: 'SFML/Graphics.hpp': No such file or directory
#include "SFML/Graphics.hpp"
         ^

X  Failed!
   Review the errors above.

```

That's all good. However, if i run the exact same `chalet build` again, without changing anything, i get:
```
󰁔 chalet build
>  Reading Build File [chalet.json] ... done (13ms)
>  Reading Microsoft (R) Visual C/C++ Environment Cache ... done (1ms)
>  Configuring build ... done (1ms)
>  Resolving source file dependencies ... done
>  C++ Compiler: Microsoft (R) Visual C/C++ version 19.37.32825 (VS 17.7.34221.43)
>  Target Architecture: x86_64-pc-windows-msvc
>  Strategy: Native
>  Configuration: Release

»  Build: cppgfx

   cppgfx: Target is up to date.

»  Build: inversekinematics.exe


X  Failed!
   Review the errors above.


ERROR: Failed to parse: build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.module.json
ERROR: There was a problem reading the json file: build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.module.json
ERROR: Parse error at line 1, column 1: syntax error while parsing value - unexpected end of input; expected '[', '{', or a literal

```
These two errors always repeat when I change the source file.
I can confirm that `build/x86_64-pc-windows-msvc_Release/obj.inversekinematics/src/main.cc.module.json` exists, but is empty.
My expectation would be that the same compiler error is shown over and over again until I fix it.

I don't know, i just have a hand for breaking software :)

**Comment by HerrNamenlos123** [comment]
I just confirmed that the error happens in a new project (`chalet init`), with MSVC and Modules enabled. And, it does not happen when there is a normal error, it only happens when a header isn't found. 

It may have to do with the fact that i used #include inside of a module, but that is valid C++20 to my knowledge.

**Comment by rewrking** [comment]
Ah, that's interesting. Definitely a bug. I'll look into it this week.

**Comment by rewrking** [comment]
This has been fixed and will be in the next release. This was kind of an MSVC bug. If the `/sourceDependencies:directives` command fails, the module.json file for that source is left blank, and so it fails to load the next time chalet is run. Now it just gets removed on failure.

**Comment by rewrking** [comment]
Available in [v0.7.8](https://github.com/chalet-org/chalet/releases/tag/v0.7.8)

**Comment by HerrNamenlos123** [comment]
This also works now. Thank you
